lib/recipes-data/src/lib/extract_recipedata_from_element.ts

```diff
@@ -0,0 +1,28 @@
+import type {Block} from "@guardian/content-api-models/v1/block";
+import {ElementType} from "@guardian/content-api-models/v1/elementType";
+
+
+export interface RecipeOutput {
+  recipeUId: string;
+  jsonBlob: string;
+  checksum?: string;
+}
+
+export function extractRecipeData(canonicalId: string, block: Block): RecipeOutput[] {
+  const allRecipes = block.elements
+    .filter(elem => elem.type === ElementType.RECIPE)
+    .map(recp => parseJsonBlob(canonicalId, recp.recipeTypeData?.recipeJson))
+  return allRecipes
+}
+
+function parseJsonBlob(canonicalId: string, recipeJson: string | undefined): RecipeOutput {
+  const recipeData = JSON.parse(recipeJson as string) as Record<string, unknown>
```

Comment:
Sorry to be a pain! I see why you've declared `recipeJson:string|undefined` in the function - to make line 14 easier - but I don't think this is right because on line 18 we are saying that it's OK for `recipeJson` to not be defined then on line 19 we are assuming that it _is_ defined.

I would change line 18 to have `recipeJson:string`, then update lines 13 & 14 to read:

```typescript
   .filter(elem => elem.type===ElementType.RECIPE && !!elem.recipeTypeData?.recipeJson)
   .map(recp => parseJsonBlob(canonicalId, recp.recipeTypeData?.recipeJson as string)
```

The reason that it's OK to do `as string` here is because we have already ensured that it _isn't_ null or undefined in the filter line.  If we're lucky then Typescript will be clever enough to realise this and we can just use `recp.recipeTypeData.recipeJson` without the `?` or forcing the type. But I suspect that it's not clever enough to realise this.

Reply:
The upshot of this change is that if we have an invalid Recipe element - which _shouldn't_ happen, but technically _can_ - then it won't break our lambda, the element will get ignored

Reply:
Yes this is making more sense, wise way of using the variable with "as string" , thanks @fredex42 for pointing it.